packages/snap/src/handlers/set-network.ts

```diff
@@ -22,15 +27,29 @@ const coerceParams = (params: SetNetworkParams): bigint => {
 };
 
 /**
- * @description Set the current network using a chain id
+ * @description Set the current network using a rpc url
  * @param params - The network parameters
- * @returns The network chain id
+ * @returns The network rpc url
  */
 export const setNetwork: Handler<SetNetworkParams, SetNetworkResponse> = async (
   params,
 ) => {
-  const network = coerceParams(params);
+  if (!params.network || typeof params.network !== 'string') {
+    throw new Error('Invalid params: network must be a string');
+  }
```

Comment:
why you dont use coerceParams ? you should use it, so every handlers looks the same

Reply:
because I don't see the point of having a function with a single line of code called in a single place